src/validation.cpp

```diff
@@ -740,7 +740,10 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
                 // Applications relying on first-seen mempool behavior should
                 // check all unconfirmed ancestors; otherwise an opt-in ancestor
                 // might be replaced, causing removal of this descendant.
-                if (!SignalsOptInRBF(*ptxConflicting)) {
+                //
+                // If replaceability signaling is ignored due to node setting,
+                // replacement is always allowed.
+                if (!SignalsOptInRBF(*ptxConflicting) && !m_pool.m_full_rbf) {
```

Comment:
nit: maybe evaluate `!m_pool.m_full_rbf` first to avoid iterating over the txins if not necessary.

Reply:
Done.